crates/experimentation_platform/src/api/experiments/helpers.rs

```diff
@@ -51,33 +51,41 @@ pub fn validate_override_keys(override_keys: &Vec<String>) -> superposition::Res
 }
 
 pub fn are_overlapping_contexts(
-    context_a: &Value,
-    context_b: &Value,
+    context_a: &Option<Value>,
+    context_b: &Option<Value>,
 ) -> superposition::Result<bool> {
-    let dimensions_a = extract_dimensions(context_a)?;
-    let dimensions_b = extract_dimensions(context_b)?;
-
-    let dim_a_keys = dimensions_a.keys();
-    let dim_b_keys = dimensions_b.keys();
-
-    let ref_keys = if dim_a_keys.len() > dim_b_keys.len() {
-        dim_b_keys
-    } else {
-        dim_a_keys
-    };
-
-    let mut is_overlapping = true;
-    for key in ref_keys {
-        let test = (dimensions_a.contains_key(key) && dimensions_b.contains_key(key))
-            && (dimensions_a[key] == dimensions_b[key]);
-        is_overlapping = is_overlapping && test;
+    match (context_a, context_b) {
+        (None, None) => Ok(true),
+        (Some(_), None) => Ok(false),
+        (None, Some(_)) => Ok(false),
+        (Some(context_a), Some(context_b)) => {
+            let dimensions_a = extract_dimensions(context_a)?;
+            let dimensions_b = extract_dimensions(context_b)?;
+
+            let dim_a_keys = dimensions_a.keys();
+            let dim_b_keys = dimensions_b.keys();
+
+            let ref_keys = if dim_a_keys.len() > dim_b_keys.len() {
+                dim_b_keys
+            } else {
+                dim_a_keys
+            };
+
+            let mut is_overlapping = true;
+            for key in ref_keys {
+                let test = (dimensions_a.contains_key(key)
+                    && dimensions_b.contains_key(key))
+                    && (dimensions_a[key] == dimensions_b[key]);
+                is_overlapping = is_overlapping && test;
+
+                if !test {
+                    break;
+                }
+            }
 
-        if !test {
-            break;
+            Ok(is_overlapping)
```

Comment:
Let's refactor this piece of code
Below code should work, though check it once.

```suggestion
           let dimensions_a = extract_dimensions(context_a)?;
            let dimensions_b = extract_dimensions(context_b)?;

            let (ref_dimensions, other_dimensions) = if dimensions_a.len() > dimensions_b.len() {
                (dimensions_b, dimensions_a)
            } else {
                (dimensions_a, dimensions_b)
            };

            let is_overlapping = ref_dimensions.iter().all(|(key, value)| {
                other_dimensions.contains_key(key) && &other_dimensions[key] == value
            });

            Ok(is_overlapping)
```

Reply:
Done